src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -420,6 +442,12 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     case Event(message: CallStarted, _) =>
       resendDueToPendingExecutionWrites(message)
       stay()
+    case Event(message @ CallCompleted(finalCallKey: FinalCallKey, _, _, _, _, _), data) if !data.isPersistedRunning(finalCallKey) =>
+      // Unlike "real" calls which are managed by a CallActor, the final call completion message is triggered
+      // internally and there isn't an automatic retry if the message isn't processed.  So if the message can't be
+      // processed due to pending writes, schedule it to be resent here.
+      resendDueToPendingExecutionWrites(message)
```

Comment:
I'd rather this be done by a FinalCallActor rather than a floating Future. And ideally, the CallStart and CallComplete would be handled by the same code block as every other call.